Gems/ScriptCanvas/Code/Include/ScriptCanvas/AutoGen/ScriptCanvasAutoGenRegistry.h

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <string>
+#include <unordered_map>
+
+#include <AzCore/RTTI/BehaviorContext.h>
+#include <AzCore/Serialization/EditContext.h>
+#include <AzCore/Serialization/SerializeContext.h>
+
+namespace ScriptCanvas
+{
+    class IScriptCanvasFunctionRegistry
+    {
+    public:
+        virtual void Reflect(AZ::ReflectContext* context) = 0;
+    };
+
+    class AutoGenRegistry
+    {
+    public:
+        AutoGenRegistry() = default;
+        ~AutoGenRegistry();
+
+        static AutoGenRegistry* GetInstance();
+        static void ReflectFunctions(AZ::ReflectContext* context);
+
```

Comment:
Have you tested this from another gem?

Reply:
not from another gem, but from project code, it works well.
But as the limitation to share static object across module, other gem/project, has to call following code manually in system component Reflect function
```
AutoGenRegistry::ReflectFunctions(context);
```